### DISCUSSION TITLE
"Failed to create prefab instance". How to use it correctly?

### DISCUSSION THREAD

**Discussion by Eugeny335** [praydog/REFramework#1112] [General Questions]
**Who can tell you how to properly spawn enemies in Resident Evil 4, outputs "Failed to create prefab instance." to the console. Maybe someone has already encountered something similar and can tell me, I will be very grateful**

using _;
using REFrameworkNET;
using via;
 
public class PrefabCache
{
    public static Dictionary<string, Prefab> prefabs = new Dictionary<string, Prefab>();
}

public class EnemySpawner
{
    private List<Prefab> prefabList;

    public EnemySpawner()
    {
        prefabList = new List<Prefab>
        {
            new Prefab("D:/REtool/re_chunk_000.pak.patch_001/natives/stm/_chainsaw/appsystem/character/ch8g2z0/ch8g2z0_body.pfb", "ch8g2z0_body"),
        };
    }

    public void SpawnEnemy(int index)
    {
        if (index < 0 || index >= prefabList.Count)
        {
            API.LogError("Invalid prefab index.");
            return;
        }

        var playerContext = GlobalService.Chainsaw._sCharacterManagerCache.getPlayerContextRef();
        if (playerContext == null)
        {
            API.LogError("Player context not found.");
            return;
        }

        vec3 playerPosition = playerContext.Position;

        Prefab prefab = prefabList[index];
        if (prefab == null || prefab.prefabManagedObject == null)
        {
            API.LogError("Prefab not found or not initialized.");
            return;
        }

        ManagedObject enemyPrefab = prefab.prefabManagedObject;

        vec3 referencePosition = REFrameworkNET.ValueType.New<vec3>();
        referencePosition.x = playerPosition.x;
        referencePosition.y = playerPosition.y;
        referencePosition.z = playerPosition.z;

        ManagedObject enemyInstance = (ManagedObject)prefab.prefabManagedObject.Call("instantiate(via.vec3)", new object[] { referencePosition });
        if (enemyInstance == null)
        {
            API.LogError("Failed to create prefab instance.");
            return;
        }

        ManagedObject enemyTransform = (ManagedObject)enemyInstance.Call("getTransform");
        if (enemyTransform == null)
        {
            API.LogError("Enemy transform not found.");
            return;
        }

        via.Quaternion enemyRotation = (via.Quaternion)enemyTransform.Call("getRotation");
        enemyTransform.Call("setPosition", new object[] { playerPosition });
        enemyTransform.Call("setRotation", new object[] { enemyRotation });

        API.LogWarning($"Enemy spawned at player's position {playerPosition} with index {index}");
    }
}

public class Prefab
{
    public ManagedObject prefabManagedObject;

    public Prefab(string path, string cacheId)
    {
        TDB tdb = TDB.Get();

        var prefabType = tdb.FindType("via.Prefab");

        var prefab = prefabType.CreateInstance(0);
        prefab.Globalize();
        prefab.Call("set_Path", new object[] { path });

        prefabManagedObject = prefab;
        PrefabCache.prefabs.Add(cacheId, this);
        API.LogWarning($"Created prefab {path}");
    }
}

**Comment by praydog** [comment]
You can't provide an absolute path to it, it needs to be relative to the game directory.
